backend/src/controllers/tournaments/put-tournament.controller..ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response } from 'express';
+import { updateTournament } from './tournament.dao';
+
+export const putTournamentController =  async(req: Request, res: Response) => {
+    const id = req.params.id;
+    const body = req.body;
+    if (body.id) {
+        res.status(405);
```

Comment:
Change this to `res.status(400).json(new BadRequest({ path: "/tournament", message: 'Changing the ID is not allowed' }));`

The 405 code refers to `MethodNotAllowed` which is about the HTTP method itself, which is not what you're trying to communicate here.